package/MDAnalysis/topology/TOPParser.py

```diff
@@ -563,9 +574,17 @@ def parsesection_mapper(self, numlines, mapper):
             A list of all entries in a given parm7 section
         """
         section = []
-        y = next(self.topfile).strip("%FORMAT(")
-        y.strip(")")
-        x = FORTRANReader(y)
+
+        def get_fmt(file):
+            if (line := next(file))[:7] == "%FORMAT":
+                return line[8:].split(")")[0]
+            else:
+                print(line)
+                return get_fmt(file)
```

Comment:
could you expand on what format you are covering here?

Reply:
Previous version assumed that `FLAG` was immediately followed by `FORMAT` and then the data. There may also be comments in the middle, so I'm skipping over those:

```
%FLAG RESIDUE_CHAINID
%COMMENT Residue chain ID (chainId) read from PDB file; DIMENSION(NRES)
%FORMAT(20a4)
```

Reply:
Arbitrary inclusions into parmed don't really count as official specification for the format in my books (parmed is great but it does a lot of things for the sake of convenience that isn't super adhering to file formats), especially given that there is a defined parm7 format spec in the amber docs.

In any case I'm not rejecting the idea of including this but a) we need to make it clear on our end via our own docs that it's not something we expect from the format, i.e. it's an additional inclusion, b) we need to work out the bounds of what arbitrary inclusions we want to have for a format that surprisingly has a definition.